core/src/main/java/com/facebook/testing/screenshot/internal/AlbumImpl.java

```diff
@@ -137,60 +110,27 @@ Bitmap getScreenshot(String name) throws IOException {
    * <p>TODO: Adjust tests to no longer use this method. It's quite sketchy and inefficient.
    */
   @Nullable
-  File getScreenshotFile(String name) throws IOException {
-    if (mZipOutputStream != null) {
-      // This needs to be a valid file before we can read from it.
-      mZipOutputStream.close();
-    }
-
-    File bundle = new File(mDir, SCREENSHOT_BUNDLE_FILE_NAME);
-    if (!bundle.isFile()) {
-      return null;
-    }
-
-    ZipInputStream zipInputStream = new ZipInputStream(new FileInputStream(bundle));
-    try {
-      String filename = getScreenshotFilenameInternal(name);
-      byte[] buffer = new byte[BUFFER_SIZE];
-
-      ZipEntry entry;
-      while ((entry = zipInputStream.getNextEntry()) != null) {
-        if (!filename.equals(entry.getName())) {
-          continue;
-        }
-
-        File file = File.createTempFile(name, ".png");
-        FileOutputStream fileOutputStream = new FileOutputStream(file);
-        try {
-          int len;
-          while ((len = zipInputStream.read(buffer)) > 0) {
-            fileOutputStream.write(buffer, 0, len);
-          }
-        } finally {
-          fileOutputStream.close();
-        }
-        return file;
-      }
-    } finally {
-      zipInputStream.close();
-    }
-    return null;
+  File getScreenshotFile(String name) {
+    return mSingleTestRunArtifactsManager.readFile(getScreenshotFilenameInternal(name));
   }
 
   @Override
   public String writeBitmap(String name, int tilei, int tilej, Bitmap bitmap) throws IOException {
     String tileName = generateTileName(name, tilei, tilej);
     String filename = getScreenshotFilenameInternal(tileName);
-    ZipOutputStream zipOutputStream = getOrCreateZipOutputStream();
-    ZipEntry entry = new ZipEntry(filename);
-    zipOutputStream.putNextEntry(entry);
-    bitmap.compress(Bitmap.CompressFormat.PNG, COMPRESSION_QUALITY, zipOutputStream);
+    ByteArrayOutputStream os = new ByteArrayOutputStream();
+    bitmap.compress(Bitmap.CompressFormat.PNG, COMPRESSION_QUALITY, os);
+    mSingleTestRunArtifactsManager.recordFile(filename, os.toByteArray());
     return tileName;
   }
 
   /** Delete all screenshots associated with this album */
   @Override
   public void cleanup() {
+    if (mCurrentTestRunId.equals(mPreviousTestRunId)) {
```

Comment:
Who will remove images after test? should it be plugin?

Reply:
That already doesn't really happen, it just "cleans up" because a new run will overwrite images in the same directory. 

We can make the plugin clear the directory on the device after pulling, yeah.